### PR TITLE
hdf5 support for afd/rd along connection

### PR DESCRIPTION
**Pull request by frheault**
This update will allow a simple cluster usage for afd/rd along connections during connectivity analysis.

For a pipeline, this leads to a conflict where the decompose.h5 needs to be in diffusion space (afd/rd along requires FODF) and in a common space for similarity analysis. So I added a `scil_apply_transform_to_hdf5.py`

Also, @mdesco requested a script to save connection to .trk in case we need some QC (not on the clusters, due to limitation of files). This was also useful to verify my code, so I added `scil_save_connections_from_hdf5.py`

Finally, I saw over the years numerous error in terms of space/header and confusion with target/reference for the `scil_apply_transformation_to_tractogram.py` and `scil_apply_warp_to_tractogram.py`. So to facilitate this I merged them into a single script where the deformation field from ants is an option (similar to `scil_transform_surface.py`)

This reduces the chance of error for an operation that always involved the two scripts together.

(These scripts are strongly related to the new **connectoflow** pipeline I am working on)

### CODE REVIEW

**Comment by GuillaumeTh** [comment]
@frheault Can you check why Travis failed?

File: docs/source/documentation/tractogram_registration.rst [hunk 1]
```diff
@@ -29,10 +29,8 @@ To apply a non-linear transformation from ANTS
 
 ::
 
-    scil_apply_warp_to_tractogram.py MOVING_FILE REFERENCE_FILE DEFORMATION_FILE OUTPUT_NAME
+    scil_apply_transform_to_tractogram.py MOVING_FILE REFERENCE_FILE  0GenericAffine.mat OUTPUT_NAME --inverse --in_deformation DEFORMATION_FILE
```
Comment (GuillaumeTh) [inline docs/source/documentation/tractogram_registration.rst:32]: Cool ! Thanks for the update

File: scilpy/utils/streamlines.py [hunk 2]
```diff
@@ -129,70 +127,90 @@ def perform_streamlines_operation(operation, streamlines, precision=None):
...
     ----------
     streamlines: list or ArraySequence
         Streamlines as loaded by the nibabel API (RASMM)
-    transfo: numpy.ndarray
-        Transformation matrix to bring streamlines from RASMM to Voxel space
-    deformation_data: numpy.ndarray
-        4D numpy array containing a 3D displacement vector in each voxel
-    source: str
-        Source of the deformation map [ants, dipy]
+    linear_transfo: numpy.ndarray
```
Comment (arnaudbore) [inline scilpy/utils/streamlines.py:141]: There are no more streamlines but sft instead

File: scilpy/utils/streamlines.py [hunk 3]
```diff
@@ -350,7 +368,7 @@ def cut_invalid_streamlines(sft):
                 logging.warning('Streamlines entirely out of the volume.')
         else:
             new_streamlines.append(sft.streamlines[ind])
-            for key_dps in sft.data_per_streamline.keys():
+            for key in sft.data_per_streamline.keys():
```
Comment (arnaudbore) [inline scilpy/utils/streamlines.py:371]: why you don't use key_dps anymore ?

File: scripts/scil_apply_transform_to_hdf5.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Transform tractogram using an affine/rigid transformation.
+
+For more information on how to use the various registration scripts
+see the doc/tractogram_registration.md readme file
```
[the remark this answers is not in the document]
Comment (arnaudbore) [reply scripts/scil_apply_transform_to_hdf5.py:8]: Still there.

File: scilpy/io/fetcher.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import logging
 import os
 import shutil
+import sys
```
Comment (arnaudbore) [inline scilpy/io/fetcher.py:7]: not used

**Comment by arnaudbore** [comment]
A lot of pep8 issues

**Pull request merged**